app/Entirety/semantics/prepDataSemantics.py

```diff
@@ -2,135 +2,253 @@
-            options.append(type)
-        return options
-
-    def relationships(self):
-        """
-        Creates a list with all possible relationships for filtering by relationship.
-        Therefore, this method itterates through the generateted df from prep_data.py.
-        Will be executeted once upon starting the App.
-        :param df:
-        :return: options (list with all possible relationships)
-        """
-        options = []
-        all_rel = []
-        for rel_list in self.df['relationship_name']:
-            for rel in rel_list:
-                if rel not in all_rel:
-                    all_rel.append(rel)
-                    options.append(rel)
-        return options
+            entity_id, entity_typ, entity_name = get_entity_dict(entity)
+            relationship_name_list, relationship_target_list = get_relationships(entity_json)
+            entity_relevant_infos_list.append(
+                [entity_id, entity_typ, entity_name, relationship_name_list, relationship_target_list])
+
+        # Build dataframe
+        entities_df = pd.DataFrame(entity_relevant_infos_list,
+                                   columns=['id', 'type', 'name', 'relationship_name', 'relationship_target'])
+        # Splits list in entities_df into new rows
+        entities_exploded_df = entities_df.copy()
+        entities_exploded_df = entities_exploded_df.explode(['relationship_name', 'relationship_target'])
+
+        # findes parents to each entity in column 'id' and appends in new column ['parents'] to entities_parents_df
+        entities_parents_df = entities_exploded_df.copy()
+        parents_dict = entities_parents_df.groupby('relationship_target')['id'].apply(list).to_dict()
+        entities_parents_df['parents'] = entities_parents_df['id'].apply(
+            lambda row: parents_dict[row] if row in entities_parents_df['relationship_target'].values else np.nan)
+        # Again split lists in column 'parents' into new ros
+        entities_parents_exploded_df = entities_parents_df.explode('parents')
+        # now each row represents an individual path from parents-> entity -> child
+
+        # Group dataframe by rows 'relationship_name', 'relationship_target' and 'parents' together, so that in
+        # now each entity is represented with its parents and children and reset index
+        entities_to_nodes_df = entities_parents_exploded_df.groupby(['id', 'type', 'name']).agg({
+            'relationship_name': list,
+            'relationship_target': list,
+            'parents': list
+        }).reset_index()
+        # Future prove code, from version 2.1.0 onwards .applymap was depreciated, it was renamed to .map
+        # Replace np.nan with []. Up until now, np.nan represented no parents, children, or relationship names.
+        # Empty lists are needed for the node generation and input format of cytoscape.js. np.nan is not accepted
+        if pd.__version__ >= '2.1.0':
```

Comment:
Instead of checking the `pandas` version, I would suggest to limit the version in [requirements.txt](https://github.com/N5GEH/n5geh.tools.entirety/blob/144-limit-search-suggestions-for-autocompletion/app/Entirety/requirements.txt). For example `pandas>=2.1.0`